shub/utils.py

```diff
@@ -136,12 +137,12 @@ def patch_sys_executable():
         # it so the system-wide Python installation uses its own library path
         # (this is particularly important if the system Python version differs
         # from the Python version that the binary was compiled with)
-        orig_lib_path = os.environ.pop(b'LD_LIBRARY_PATH', None)
+        orig_lib_path = os.environ.pop('LD_LIBRARY_PATH', None)
```

Comment:
> Would it be enough to wrap it in to_native_str()?

Well, yes, but you only need to do that for variables that may come in binary or unicode. Literals without any prefix should already be of "native" type.